app/controllers/interstitial_controller.rb

```diff
@@ -19,6 +19,12 @@ def show
   def redirect_param_same_as_host?
     return false if params[:redirect_to].blank?
 
-    URI.parse(URI.decode(params[:redirect_to])).host == request.host
+    param_host = begin
+                   URI.parse(URI.decode(params[:redirect_to]))
+                 rescue URI::InvalidURIError
+                   URI.parse(params[:redirect_to])
```

Comment:
TBH, I'm not 100% sure what `URI.decode` is trying to guard us against here, but I thought this would be the best approach that would theoretically not break other cases that I'm not thinking about.  URI parsing always confuses me, so if folks have better/safer ways of getting at this data, I'm all ears.